File: lib/core_services/interfaces/paginate.dart
class Paginate<T> {
  final List<T> items;
  final int page;
  final int totalItems;
  final int totalPages;

  const Paginate({
    required this.items,
    required this.page,
    required this.totalItems,
    required this.totalPages,
  });

  factory Paginate.createFakeData(List<T> items) {
    return Paginate(
      items: items,
      page: 1,
      totalItems: items.length,
      totalPages: 1,
    );
  }
}

File: lib/core_services/interfaces/paginate.ts
export interface Paginate<T> {
  items: T[]
  page: number
  totalItems: number
  totalPages: number
}

export function createFakePaginate<T>(item: T): Paginate<T> {
  return {
    items: Array(30).fill(item),
    page: 1,
    totalItems: 30,
    totalPages: 1
  }
}
